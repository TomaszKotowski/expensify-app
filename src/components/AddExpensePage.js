import React from 'react';
import ExpenseForm from './ExpenseForm'
const AddExpensePage = () => (
    <div>
        <h1>Add expense</h1>
        <ExpenseForm />
    </div>
);

export default AddExpensePage;