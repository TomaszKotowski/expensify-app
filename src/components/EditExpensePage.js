import React from 'react';
const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            Editing the expenses with id of {props.match.params.id}
        </div>
    );
};

export default EditExpensePage;