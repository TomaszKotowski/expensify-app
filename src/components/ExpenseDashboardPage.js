import React from 'react';
import ExpenseList from './ExpenseList'
import ExpneseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpneseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;