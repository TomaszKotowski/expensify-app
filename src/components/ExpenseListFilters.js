import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filtes';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChanged = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    };
    onTextChange = (event) => {
        this.props.setTextFilter(event.target.value);
    };
    onSortChange = (event) => {
        if (event.target.value === 'date') {
            this.props.sortByDate();
        } else if (event.target.value === 'amount') {
            this.props.sortByAmount();
        }
    };
    render() {
        return (
            <div>
                <input
                    type='text'
                    value={this.props.filters.text}
                    onChange={this.onTextChange} />
                <select
                    value={this.props.filters.sortBY}
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChanged}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter()),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters); 