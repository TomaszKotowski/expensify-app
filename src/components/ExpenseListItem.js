import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`} >
            <h3>{description}</h3>
        </Link>
        <p>Amount: {amount}</p>
        <p>Created at: {createdAt}</p>
    </div>
);

export default ExpenseListItem;