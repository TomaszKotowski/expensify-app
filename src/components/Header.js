import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <NavLink exact to="/" activeClassName="is-active">Dashboard</NavLink>
            <NavLink exact to="/create" activeClassName="is-active">Create Expense</NavLink>
            <NavLink exact to="/help"activeClassName="is-active">Help</NavLink>
        </nav>
    </header>
);

export default Header