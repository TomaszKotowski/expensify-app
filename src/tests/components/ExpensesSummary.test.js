import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('should correctly render ExpeneseSummmary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpeneseSummmary with mulitple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2318494985} />);
    expect(wrapper).toMatchSnapshot();
});