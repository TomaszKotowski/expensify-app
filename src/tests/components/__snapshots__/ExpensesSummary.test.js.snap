// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpeneseSummmary with mulitple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    23,184,949.85 $
  </h1>
</div>
`;

exports[`should correctly render ExpeneseSummmary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    2.35 $
  </h1>
</div>
`;
